File: games/coinFlip.c
#include <stdio.h>
#include <stdlib.h>
#include<string.h>
#include<time.h>

int main(){
  
  int a=0,b=0;
  char c[7]="";
  
  printf("Heads or Tails?");
  scanf("%s",c);
  
  srand(time(0));
  b= (rand() % 2);
  
  if(strcmp(c,"heads") == 0){
    a=1;
    if(a==b){printf("Correct\n");}
    else {printf("Incorrect\n");}
  }
    else {a=0;printf("Incorrect\n");}
  
  while(strcmp("done",c) != 0) {    
    printf("Type \"done\" to quit.\n\n");
    printf("Heads or Tails?");
    scanf("%s",c);
    
    b= (rand() % 2);
    
    if(c=="Heads"||"heads"||"head"||"heads"){
	a=1;
	if(a==b){printf("Correct\n");}
	else {printf("Incorrect\n");}
    }
    if (strcmp(c,"done")) {exit(0);}
    else {a=0;printf("Incorrect\n");}
    
  }
  
  return 0;
}

File: games/coinFlip.java
package testingGrounds;

import java.util.Scanner;
import java.util.Random;

public class coinFlip {

	public static void main(String[] args) {

		Scanner input = new Scanner(System.in);
		Random number = new Random();

		int b,a=0;
		String c;
		b = number.nextInt(2);

		System.out.println("Heads or Tails?");
		c = input.next();
		if (c.equals("Heads")||c.equals("heads")||c.equals("h")) {a=1;}
		else {a=0;}
		
		if (a==b) {
			System.out.println("Correct!\n");
		  }
		else {
			System.out.println("Incorrect!\n");
		  }
		
		while (!c.equals("done")) {
		
		b = number.nextInt(2);
		
		System.out.println("Type \"done\" to stop");
		System.out.println("Heads or Tails?");
		c = input.next();
		if (c.equals("Heads")||c.equals("heads")||c.equals("h")) {a=1;}
		else if (c.equals("done")) {break;}
		else {a=0;}
		
		if (a==b) {
			System.out.println("Correct!\n");
		  }
		else {
			System.out.println("Incorrect!\n");
		  }
		}
		input.close();
	}
}
